### ISSUE TITLE
CO Production Measurement

### ISSUE THREAD

**Issue by gforney** [firemodels/fds#438]
```
Application Version: 5.1.6
SVN Revision Number: 5.1.6
Compile Date:
Operating System: Windows XP

Describe details of the issue below:

We are looking for a way to measure CO production through a surface, or out
of a volume. We would like to eventually arrive at kg/s or mol/s.

Using devices...

I have created a point CO concentration device in the center of the plane
in question for reference. If you use the device line...

&DEVC ID='CO CONC AT PLANE', QUANTITY='carbon monoxide',
XB=-0.0540,0.1080,0.00,0.0800,0.1480,0.1480 /

... you get CO concentration measurements identical to those from the point
CO concventration device at the center of the plane. I was referred to
section 12.2.3 in the FDS5 user guide on output statistics. So maybe I
should use...

&DEVC ID='CO CONC AT PLANE', QUANTITY='carbon monoxide',
XB=-0.0540,0.1080,0.00,0.0800,0.1480,0.1480 , STATISTICS=’AREA INTEGRAL’/ 

...however, the simulation won't even run with this line. I get "ERROR:
problem with DEVC no.2". So maybe my syntax is wrong, or maybe the output
statistics isn't suitable for this DEVC operation.

The Sample fds file is attached.Either way, I am going to be getting an
average over the entire plane, and multiplying by another average of the
flow velocity over the entire plane. What I really need is to have the
concentration multiplied by the flow velocity at each node at each timestep
to get at something like average kg/s or mol/s of CO production, or even
total grams or mols.

Another idea is to create a CO concentration and flow velocity slice. Then
parse the files using the fds2ascii. The import to something like matlab.
In matlab multiply the CO concentration by the flow velcocity to get mol/s.
Two problems, 1.) CO concentration is usually such a small number that we
are worried about roundoff 2.) The larger problem is that using the
fds2ascii file requires time averaging of the two quantities independently.

Thanks, any suggestions, ideas or corrections are greatly appreciated.

```

Original issue reported on code.google.com by `jagenbro` on 2008-09-04 16:57:01

<hr>
- _Attachment: [GEN3.fds](https://storage.googleapis.com/google-code-attachments/fds-smv/issue-459/comment-0/GEN3.fds)_

**Comment by gforney** [comment]
```
You have the wrong type of quote character surrounding AREA INTEGRAL.  Should be the

same as the type of quote surrounding the QUANTITY.  
```

Original issue reported on code.google.com by `drjfloyd` on 2008-09-04 19:01:04
- **Labels added**: OpSys-All, Application-FDS

**Comment by gforney** [comment]
```
that was easy enough, now it runs. But what about getting a mass flux at this plane
instead of mol/mol concentration. I am reading something similar under the 12.2.3
p.104 of the fds user guide under area integrals. Their is some mention of a mass
flux, but I am unclear about how to code this, and if it will in fact give me a mass
flux of CO at a surface in kg/s
```

Original issue reported on code.google.com by `jagenbro` on 2008-09-04 19:18:09

**Comment by gforney** [comment]
```
The quote problem is not an uncommon error, but one that can be hard to notice

If for your 2nd DEVC you give the QUANTITY='MASS FLUX Z'  and 
SPEC_ID='MIXTURE_FRACTION_2' you would get the mass flux of the second mixture 
fraction species across the plane you defined with XB.  You would then need to 
convert that to CO mass using the formulas in the Technical Manual.  At this point

in time we haven't yet added the logic in the code to support extracting specific 
species from the mixture fraction using MASS FLUX.
```

Original issue reported on code.google.com by `drjfloyd` on 2008-09-04 19:39:59

**Comment by gforney** [comment]
```
Thank you so much drjfloyd,

sorry to take up so much of your time on this one issue, but I feel like we are
getting close, and this would help our simulation greatly. My DEVC line now looks
like this...

&DEVC ID='CO CONC AT PLANE', QUANTITY='MASS FLUX Z',
XB=-0.0540,0.1080,0.00,0.0800,0.1480,0.1480 , STATISTICS='AREA INTEGRAL',
SPEC_ID='MIXTURE_FRACTION_2'/

and I am getting a ERROR: problem with DEVC no. 2 message once again. I have attached
the fds file as well. I am unclear as to how the SPEC_ID is used.

Thanks again for all of your help and patience.
```

Original issue reported on code.google.com by `jagenbro` on 2008-09-04 20:49:51

<hr>
- _Attachment: [GEN3.fds](https://storage.googleapis.com/google-code-attachments/fds-smv/issue-459/comment-4/GEN3.fds)_

**Comment by gforney** [comment]
```
You'll need to upgrade to FDS 5.2 to make use of that DEVC.  The DEVC will integrate

the MASS FLUX of the species identified by SPEC_ID.  FDS does not track CO, CO2, 
H2O, etc.  It tracks the mixture fraction variables.  In FDS 5.2 with CO_PRODUCTION

set to .TRUE. there are three of them.  MIXTURE_FRACTION_1, MIXTURE_FRACTION_2, 
MIXTURE_FRACTION_3.  These variables can be converted back to CO, CO2, etc. by the

use of the state relations discussed in the Theory Manual.  I know right now this is

a bit of a pain, but we do plan to extend the capability to be able to specifiy one

of the mixture fraction tracked species like 'carbon monoxide'.
```

Original issue reported on code.google.com by `drjfloyd` on 2008-09-04 21:12:57

**Comment by gforney** [comment]
```
FYI -- this has been on my to-do list. Jason is giving you a backdoor method to get

what you need, but I plan to make this easier. In the next release, we'll make it so

that you can get a mass flux of carbon monoxide or any other mixture fraction 
variable.
```

Original issue reported on code.google.com by `mcgratta` on 2008-09-05 12:39:17

**Comment by gforney** [comment]
```
Now that I have the devices in place, I would like to check my understanding of how
to back out the CO production numbers.

From the technical reference manual, P.26...

Y_CO = v_CO*(W_CO/W_F)*InitialY_F*Z_2                   EQ (3.44)

substituting equation (3.48_ for v_CO, the above equation becomes...

Y_CO=y_CO*InitialY_F*Z2

My first question is where do I obtain InitialY_F. The best Idea that I have is to
look at a fuel concentration slice animation to get X_F, then convert to mass/mass.
Is the parameter defined somewhere in the input? does it change with burner
intensity? we are using a solid pyrolysis specifying A and E as the source of our
combustion gasses.

The Z2 mass flux is being output as kg/s/m^2. My second question is do I multiply by
the cell volume size or the XB device area to get to kg/s. Or neither? Basically is
the area integral and average or a summation.

Thanks
```

Original issue reported on code.google.com by `jagenbro` on 2008-09-09 18:32:54

**Comment by gforney** [comment]
```
YsubFsuperI is defined below equation 3.24 as the fuel mass fraction at the burner 
surface.  This is 1 in your case.  

By selecting AREA INTEGRAL for the statistics your answer is in kg/s
```

Original issue reported on code.google.com by `drjfloyd` on 2008-09-09 18:47:54

**Comment by gforney** [comment]
```
I was hoping for kg/s, but when I open the Device .CSV file, the column header gives
units of kg/s/m^2. I attached the file as well.

So to be 100% clear,

if my CO yield (y_CO) is 0.03, then my CO production will be...

0.03*1*Z2 which gives an answer in the units of Z2, hopefully kg/s.

Thanks
```

Original issue reported on code.google.com by `jagenbro` on 2008-09-09 18:54:53

<hr>
- _Attachment: [GEN3.fds](https://storage.googleapis.com/google-code-attachments/fds-smv/issue-459/comment-9/GEN3.fds)_

**Comment by gforney** [comment]
```
selecting STATISTICS of AREA INTEGRAL results in units of kg/s not kg/m^2/s.  The 
column heading is no longer correct in this case.  Other than that,  yes (area 
integral of the mass flux of Z2) * CO_YIELD will give you CO in kg/s
```

Original issue reported on code.google.com by `drjfloyd` on 2008-09-09 19:01:19

**Comment by gforney** [comment]
```
The units do not get converted when you use AREA_INTEGRAL. Something for the to-do 
list.
```

Original issue reported on code.google.com by `mcgratta` on 2008-09-09 19:01:35

**Comment by gforney** [comment]
```
I added the ability to get a mass flux of a mixture fraction species. Should be 
working in 5.2.1
```

Original issue reported on code.google.com by `mcgratta` on 2008-09-11 22:10:01

**Comment by gforney** [comment]
```
Have you tested the fix since the release of 5.2.4?  If so please report back if the
fix has addressed your issue.
```

Original issue reported on code.google.com by `drjfloyd` on 2008-12-03 14:13:27

**Comment by gforney** [comment]
```
(No text was entered with this change)
```

Original issue reported on code.google.com by `drjfloyd` on 2009-03-13 15:23:00
